src/squlearn/util/pennylane/pennylane_circuit.py

```diff
@@ -0,0 +1,509 @@
+    #         }
+    #     )  #
+    #     modules = jnp
+    # elif self._gradient_engine == "torch" or self._gradient_engine == "pytorch":
+    #     from sympy.printing.pycode import PythonCodePrinter as Printer  # type: ignore
+
+    #     user_functions = {}
+    #     printer = Printer(
+    #         {
+    #             "fully_qualified_modules": False,
+    #             "inline": True,
+    #             "allow_unknown_functions": True,
+    #             "user_functions": user_functions,
+    #         }
+    #     )  #
+    #     modules = torch
+
+    # else:
+    #     # tbd for jax and tensorflow
+    #     printer = None
+    #     modules = None
+
+    return printer, modules
+
+
+class PennyLaneCircuit:
+    """
+    Class for converting a Qiskit circuit to a PennyLane circuit.
+
+    Args:
+        circuit (QuantumCircuit): Qiskit circuit to convert to PennyLane
+        observable (Union[None, SparsePauliOp, List[SparsePauliOp], str]): Observable to be measured
+                                                                           Can be also a string like ``"probs"`` or ``"state"``
+        executor (Executor): Executor object to handle the PennyLane circuit. Has to be initialized with a PennyLane device.
+
+    Attributes:
+    -----------
+
+    Attributes:
+        pennylane_circuit (qml.qnode): PennyLane circuit that can be called with parameters
+        circuit_parameter_names (list): List of circuit parameter names
+        observable_parameter_names (list): List of observable parameter names
+        circuit_parameter_dimensions (dict): Dictionary with the dimension of each circuit parameter
+        observable_parameter_dimension (dict): Dictionary with the dimension of each observable parameter
+        circuit_arguments (list): List of all circuit and observable parameters names
+        hash (str): Hashable object of the circuit and observable for caching
+
+    Methods:
+    --------
+    """
+
+    def __init__(
+        self,
+        circuit: QuantumCircuit,
+        observable: Union[None, SparsePauliOp, List[SparsePauliOp], str] = None,
+        executor: Executor = None,
+    ) -> None:
+
+        self._executor = executor
+        # Transpile circuit to statevector simulator basis gates to expand blocks automatically
+        self._qiskit_circuit = transpile(circuit, backend=Aer.get_backend("statevector_simulator"))
+
+        self._qiskit_observable = observable
+        self._num_qubits = self._qiskit_circuit.num_qubits
+
+        if self._executor:
+            if self._executor.quantum_framework != "pennylane":
+                raise ValueError(
+                    "Executor is not compatible with PennyLane. Please provide a executor initialized with a PennyLane device!"
+                )
+            self._decorator = self._executor.pennylane_decorator
+        else:
+            device = qml.device("default.qubit")
+
+            def pennylane_decorator(pennylane_function):
```

Comment:
1. also here don't like the name
2. also here, why isn't this just a member function?

Reply:
Done